fix: prevent empty availability schedule names

xDipzz opened the pull request:
## What does this PR do?

- Add frontend validation with error toast and i18n support
- Add backend validation with `.trim().min(1)` in Zod schemas
- Add i18n translation for `schedule_name_cannot_be_empty`
- Add comprehensive tests for validation (9 tests, all passing)
- Sync `previousValue` with prop changes in EditableHeading
- Follow PR #24044 approach with all reviewer feedback addressed

Fixes #21709

## Visual Demo (For contributors especially)

### Before (Bug)

https://github.com/user-attachments/assets/aba0a5d3-141b-45bf-b60c-f7792ac8bc68



### After (Fixed)

https://github.com/user-attachments/assets/c6b83737-fe51-4810-9a96-22cd12948411



## Tests Added

**create.schema.test.ts** - 4 tests covering:
- Valid schedule names
- Trimming whitespace
- Rejecting empty strings
- Rejecting whitespace-only strings

**update.schema.test.ts** - 5 tests covering:
- Valid schedule names
- Trimming whitespace  
- Rejecting empty strings
- Rejecting whitespace-only strings
- Optional name handling

All 9 tests passing

## Approach (Following PR #24044)

### Frontend
- Shows error toast when user tries to save empty name
- Uses i18n for multi-language support
- Silent auto-revert in EditableHeading for better UX
- Syncs previousValue with prop changes to prevent stale state

### Backend
- Zod schemas use `.trim().min(1)` validation
- Prevents empty/whitespace-only names at API level
- Consistent error messages

### Reviewer Feedback Addressed
- @coderabbitai: Added trimming + min-length check to schemas
- @volnei: Added comprehensive tests
- @Pallava-Joshi: Fixed i18n translation properly
- cubic-dev-ai: Fixed previousValue sync issue in EditableHeading

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a documentation change. N/A - No documentation changes needed
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works

## How should this be tested?

1. Go to Availability and click on any availability schedule
2. Click on edit heading
3. Delete all text or type only spaces
4. Click save
5. Expected: Error toast appears "Schedule name cannot be empty"
6. Run tests: `yarn test create.schema.test.ts update.schema.test.ts`

## Checklist

- [x] I have read the contributing guide
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have checked if my changes generate no new warnings

xDipzz committed:
fix: prevent empty availability schedule names

- Add validation to revert to previous value if name is empty
- Silent auto-revert without error messages (works for all languages)
- Add backend safety net with Zod transform to default value
- Non-breaking API changes as requested by maintainers

Fixes #21709

dhairyashiil reviewed (changes requested):
Can you refer to a similar approach as in PR #24044 and make sure to add tests as well? Please also consider the reviewer’s comments from that PR. Thanks.

xDipzz committed:
fix: prevent empty availability schedule names

- Add frontend validation with error toast and i18n support
- Add backend validation with .trim().min(1) in Zod schemas
- Add i18n translation for schedule_name_cannot_be_empty
- Add comprehensive tests for create and update schemas (9 tests)
- Sync previousValue with prop changes in EditableHeading
- Follow PR #24044 approach with all reviewer feedback addressed

Fixes #21709

xDipzz committed:
fix: correct Tailwind class typo in AvailabilitySettings

xDipzz committed:
fix: remove useEffect that breaks previousValue logic

xDipzz committed:
Merge branch 'main' into fix/prevent-empty-schedule-name

xDipzz committed:
fix: remove useEffect after merge

xDipzz committed:
fix: update tests and UI components for schedule validation

xDipzz committed:
fix: use optional chaining for onValueChange callback

xDipzz committed:
fix: sync previousValue with value prop changes

Ryukemeister reviewed (commented):
hi @xDipzz , thank you for creating this PR. although the approach you took is totally valid, I have a few suggestions. lemme know what you think about it

Ryukemeister commented on packages/trpc/server/routers/viewer/availability/schedule/update.schema.ts:
```diff
@@ -5,7 +5,7 @@ import { timeZoneSchema } from "@calcom/lib/dayjs/timeZone.schema";
   scheduleId: z.number(),
   timeZone: timeZoneSchema.optional(),
-  name: z.string().optional(),
+  name: z.string().trim().min(1, "Schedule name cannot be empty").optional(),
```
its better to replace the actual text with translations

devin-ai-integration[bot] committed:
merge: resolve conflicts with upstream main

Co-Authored-By: unknown <>

sahitya-chandra commented:
Closing this pr as the linked issue is already fixed
